administrative-sdk/pronunciationAnalysis.js

```diff
@@ -230,246 +215,169 @@ class PronunciationAnalysis {
+        connection._analysisId = null;
+
+        // When done, submit any plain text (non-JSON) to start analysing.
+        connection._session.call('nl.itslanguage.pronunciation.analyse',
+          [connection._analysisId], {}, {receive_progress: true})
+          .then(reportDone)
+          .catch(function(res) {
+            if (res.error === 'nl.itslanguage.ref_alignment_failed') {
+              res.kwargs.analysis.message = 'Reference alignment failed';
+            } else if (res.error === 'nl.itslanguage.alignment_failed') {
+              res.kwargs.analysis.message = 'Alignment failed';
+            } else if (res.error === 'nl.itslanguage.analysis_failed') {
+              res.kwargs.analysis.message = 'Analysis failed';
+            } else {
+              res.kwargs.analysis.message = 'Unhandled error';
+              Connection.logRPCError(res);
+            }
+            reportError(res.kwargs.analysis);
+          })
+          .tap(function(progress) {
+            reportProgress(progress);
+          });
+      };
+
+      recorder.addEventListener('recorded', stopListening);
+      connection._session.call('nl.itslanguage.pronunciation.init_analysis', [],
+        {
+          trimStart: trimAudioStart,
+          trimEnd: trimAudioEnd
+        })
+        .then(initAnalysis)
+        .then(() => {
+          self.pronunciationAnalysisInitChallenge(connection, challenge)
+            .then(function() {
+              var p = new Promise(function(resolve) {
+                if (recorder.hasUserMediaApproval()) {
+                  resolve();
+                } else {
+                  recorder.addEventListener('ready', resolve);
+                }
+              });
+
+              p.then(function() {
+                recorder.removeEventListener('ready', resolve);
+                self.pronunciationAnalysisInitAudio(connection, recorder, startStreaming);
+              });
+            });
+        })
+        .catch(Connection.logRPCError);
+    });
   }
 
   /**
-   * Callback used by getPronunciationAnalysis.
-   *
-   * @callback Sdk~getPronunciationAnalysisCallback
-   * @param {its.PronunciationAnalysis} analysis Retrieved pronunciation analysis domain model instance.
-   */
-  getPronunciationAnalysisCallback(analysis) {}
-
-  /**
-   * Error callback used by getPronunciationAnalysis.
-   *
-   * @callback Sdk~getPronunciationAnalysisErrorCallback
-   * @param {object[]} errors Array of errors.
-   */
-  getPronunciationAnalysisErrorCallback(errors) {}
-
-  /**
    * Get a pronunciation analysis in a pronunciation challenge.
    *
    * @param {PronunciationChallenge} challenge Specify a pronunciation challenge.
    * @param {string} analysisId Specify a pronunciation analysis identifier.
    * @param {Sdk~getPronunciationAnalysisCallback} [cb] The callback that handles the response.
    * @param {Sdk~getPronunciationAnalysisErrorCallback} [ecb] The callback that handles the error response.
```

Comment:
Missing documentation update. Callbacks are no longer taken and a promise is returned.